pkg/processor/procjs/jsprocessor.go

```diff
@@ -0,0 +1,184 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procjs
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/processor"
+	"github.com/conduitio/conduit/pkg/record"
+	"github.com/dop251/goja"
+	"github.com/rs/zerolog"
+)
+
+const (
+	entrypoint = "process"
+)
+
+// JSProcessor is able to run processors defined in JavaScript.
+type JSProcessor struct {
```

Comment:
Nitpick: given that it's the only processor in this package I'd just name it `Processor` and the file `processor.go`.